src/send_proc.py

```diff
@@ -15,148 +15,147 @@
 chore_struct = utils.readStructFromJSON("chores.json")
 
 
-def run() -> None:
-    hasUpdated = notCompleted = False 
-    violationSent = True
-    notCompleted = True
-    isViolationTime = lambda time: time == 0 or time % 15 == 0 or time % 30 == 0 or time % 45 == 0
+def runSchedule() -> None:
+	global assigned
+	lazies: list = None
+	hasUpdated = notCompleted = violationSent = punishTime = False 
+	isViolationTime = lambda time: time == 0 or time % 15 == 0 or time % 30 == 0 or time % 45 == 0
+	isDailyUpdateTime = lambda now: now.hour == 8 and now.minute == 0
+	choresArentDone = lambda lazies: len(lazies) != 0 
 
-    while True:
-        now = datetime.datetime.now()
 
+	while True:
 
-        if now.hour == 8 and now.minute == 0 and hasUpdated is False: #time to build & assign today's chores 
-            hasUpdated, notCompleted = True, True 
-            global assigned
-            assigned = generateChoreAssignments()
-            hasUpdated = False 
+		now = datetime.datetime.now()
 
-        if not isViolationTime(now.minute):
-            violationSent = False 
+		if hasUpdated and not isViolationTime(now):
```

Comment:
Would be nice to have type specificity for `isVolationTime`... You're passing in a datetime object when the function expects a minute value

Reply:
:: +1 ::

Reply:
fc8506f